library/python/src/pipeline.cc

```diff
@@ -0,0 +1,61 @@
+#include "pywavemap/pipeline.h"
+
+#include <nanobind/stl/shared_ptr.h>
+#include <nanobind/stl/string.h>
+#include <nanobind/stl/vector.h>
+#include <wavemap/pipeline/pipeline.h>
+
+using namespace nb::literals;  // NOLINT
+
+namespace wavemap {
+void add_pipeline_bindings(nb::module_& m) {
+  nb::class_<Pipeline>(m, "Pipeline",
+                       "A class to build pipelines of measurement integrators "
+                       "and map operations.")
+      .def(nb::init<std::shared_ptr<MapBase>>(), "map"_a)
+      .def("clear", &Pipeline::clear,
+           "Deregister all the pipeline's measurement integrators and map "
+           "operations.")
+      .def("hasIntegrator", &Pipeline::hasIntegrator, "integrator_name"_a,
```

Comment:
Would be `has_integrator` or alternatively `integrator_name in pipeline`. The `in` keyword is used in python to check for existence of an object in a different thing, often lists, maps, etc. So might not be the most sensible here to be honest unless you return a list of integrators but then it's on the user to implement the check.